app/controllers/citizens_controller.rb

```diff
@@ -0,0 +1,79 @@
+class CitizensController < ApplicationController
+  before_action :set_states, only: [:new, :edit]
+  def index
+    @citizens = Citizen.all.order(:id)
+  end
+
+  def new
+    @citizen = Citizen.new
+    @citizen.build_address
+  end
+
+  def create
+    @citizen = Citizen.new(permitted_params)
+    respond_to do |format|
+      if @citizen.save
+        format.html { redirect_to root_path, notice: "Incluido com sucesso." }
+      else
+        format.html { render :new, error: "Erro ao incluir", status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    @citizen = Citizen.find_by_id(params[:id])
+  end
+  
+  def update
+    @citizen = Citizen.find(params[:id])
+    if @citizen.update_attributes(permitted_params)
+      flash[:success] = "Object was successfully updated"
+      redirect_to citizens_path
+    else
+      flash[:error] = "Something went wrong"
+      render :edit
+    end
+  end
+  
+  
+  private
+
+  def permitted_params
+    params.require(:citizen)
+    .permit(:id, :name, :cpf, :cns, :photo, :birth_date, :phonenumber, :status, address_attributes: [
+      :id, :zipcode, :ibge_code, :address, :street, :complement, :neighborhood, :city, :state, :_destroy
+    ] )
+  end
+
+  def set_states
+    @states = [
```

Comment:
This is a BAD ideia. Colocar dados estáticos dentro de um controller? :trollface: 

Você poderia ser simplista e mover esse método para um helper, e daí chamá-lo de vários lugares. Esse seria o mínimo.

Idealmente, somando ao mínimo tente utilizar rails caching: https://guides.rubyonrails.org/caching_with_rails.html